hedera-node/src/main/java/com/hedera/services/grpc/marshalling/AliasResolver.java

```diff
@@ -225,24 +223,70 @@ private Result resolveInternalFungible(
             }
             final var resolution = aliasManager.lookupIdBy(alias);
             if (resolution == MISSING_NUM) {
-                result = netOf(isEvmAddress, alias);
+                if (isForToken) {
+                    result = netOf(isEvmAddress, alias, false);
+                } else {
+                    result = netOf(isEvmAddress, alias, true);
+                }
             } else {
                 resolvedAdjust =
                         adjust.toBuilder().setAccountID(resolution.toGrpcAccountId()).build();
             }
             resolutions.put(alias, resolution);
+            tokenTransferResolutions.put(alias, resolution);
         }
         resolvingAction.accept(resolvedAdjust);
         return result;
     }
 
-    private Result netOf(final boolean isEvmAddress, final ByteString alias) {
+    private void perceiveNftReceiverResult(final Result result, final NftTransfer change) {
+        perceiveResult(
+                result, change.getSerialNumber(), change.getReceiverAccountID().getAlias(), false);
+    }
+
+    private void perceiveNonNftResult(final Result result, final AccountAmount adjust) {
+        perceiveResult(result, adjust.getAmount(), adjust.getAccountID().getAlias(), true);
+    }
+
+    private void perceiveResult(
+            final Result result,
+            final long assetChange,
+            final ByteString alias,
+            final boolean repetitionsAreInvalid) {
+        if (result == Result.UNKNOWN_ALIAS) {
+            if (assetChange > 0) {
+                if (isSerializedProtoKey(alias)) {
+                    perceivedCreations++;
+                } else {
+                    perceivedInvalidCreations++;
+                }
+            } else {
+                perceivedMissing++;
+            }
+        } else if (repetitionsAreInvalid && result == Result.REPEATED_UNKNOWN_ALIAS) {
+            perceivedInvalidCreations++;
+        } else if (result == Result.UNKNOWN_EVM_ADDRESS) {
+            perceivedMissing++;
+        }
+    }
+
+    private Result netOf(
+            final boolean isEvmAddress, final ByteString alias, final boolean isForNftOrHbar) {
         if (isEvmAddress) {
             return Result.UNKNOWN_EVM_ADDRESS;
-        } else {
+        } else if (isForNftOrHbar) {
+            // Note a REPEATED_UNKNOWN_ALIAS is still valid for the NFT receiver case
             return resolutions.containsKey(alias)
                     ? Result.REPEATED_UNKNOWN_ALIAS
                     : Result.UNKNOWN_ALIAS;
+        } else {
+            // If the token resolutions map already contains this unknown alias, we can assume
```

Comment:
I know I suggested this comment, but I can't understand it at all any more, so might as well delete I guess. 🤷

Reply:
haha I still understand it 😝 . I think you meant , if the alias is repeated in the previous change in transferList, it would have gone through `autoCreation` already, so it need not auto create again.